hetmech/dwpc.py

```diff
@@ -0,0 +1,108 @@
+import functools
+import operator
+
+import numpy
+
+from .matrix import metaedge_to_adjacency_matrix, normalize, copy_array
+
+
+def remove_diag(mat):
+    """Set the main diagonal of a square matrix to zeros."""
+    assert len(set(mat.shape)) == 1  # must be square
+    return mat - numpy.diag(mat.diagonal())
```

Comment:
I thought we had decided to always subtract a sparse diagonal for performance / minimizing RAM? Were there type issues? Just a note, can deal with this later.